### PR TITLE
Allow pasting new fragment

### LINKED ISSUE #50: Pasting a new fragment does not initiate a new session
When pasting a URL with a new fragment to an existing browser window, state is not updated to take the new fragment into account. A reload is required to initiate the connection.

### PR DESCRIPTION
**Pull request by lxgr**
This should fix #50.

### CODE REVIEW

**Comment by lxgr** [comment]
Hm, this does not work for a portal that was never opened. Let me see if I can invert the logic (only ignore the event if we change the fragment ourselves)

**Comment by lxgr** [comment]
This should do it. Currently running here: https://43fbe5da.ngrok.io/

File: web/main.js [hunk 1]
```diff
@@ -233,7 +235,9 @@ let disconnected = () => {
 	document.body.removeEventListener('drop', unhighlight);
 	document.body.removeEventListener('dragleave', unhighlight);
 
+	window.removeEventListener('hashchange', hashchange)
 	location.hash = "";
+	window.addEventListener('hashchange', hashchange)
```
Comment (saljam) [inline web/main.js:240]: given that we check that it's not empty in `joining` do we need to remove and add it?

File: web/main.js [hunk 2]
```diff
@@ -249,6 +253,18 @@ let preventdefault = e => {
 	e.stopPropagation()
 }
 
+let joining = () => {
+	document.getElementById("magiccode").value = location.hash.substring(1);
+		document.getElementById("dial").value = "JOIN WORMHOLE";
+		connect();
```
Comment (saljam) [inline web/main.js:259]: i think there's a couple of spare tabs in these lines :)

File: web/main.js (hunk 1 above)
Comment (saljam) [reply web/main.js:240]: (if there are no issues with that let's move `window.addEventListener('hashchange', hashchange)` to the `DOMContentLoaded` block at the bottom of this file with the other handler initialisations code)

Comment (lxgr) [reply web/main.js:240]: I was running into a loop somewhere, let me see if I can fix this another way.

File: web/main.js (hunk 2 above)
Comment (lxgr) [reply web/main.js:259]: True, this used to be an if statement ;) Removing!

File: web/main.js (hunk 1 above)
Comment (lxgr) [reply web/main.js:240]: Heh, now I'm connecting with my own instance, so we need to at least also check that it has in fact changed due to user input and not our own programmatic change when initiating.

**Comment by lxgr** [comment]
Think I got it now – let me know what you think of this approach. Removing and adding the event handler only around setting the hash to the newly generated code would seem to do the trick as well.

**Pull request merged**